pkg/operator/hive/hive.go

```diff
@@ -513,3 +514,47 @@ func (r *ReconcileHiveConfig) deleteAllSyncSetInstances(hLog log.FieldLogger) (n
 	}
 	return
 }
+
+func (r *ReconcileHiveConfig) deployMonitoring(hLog log.FieldLogger, h resource.Helper, instance *hivev1.HiveConfig) error {
+	hiveNSName := getHiveNamespace(instance)
+
+	if !instance.Spec.ExtractMetrics {
+		hLog.Info("metrics were not requested to be extracted, so skipping")
+		return nil
+	}
+
+	// setup the hiveNSName labels so that openshift-prometheus discovers the namespace.
+	ns := &corev1.Namespace{}
+	if err := r.Client.Get(context.TODO(), types.NamespacedName{Name: hiveNSName}, ns); err != nil {
+		hLog.WithError(err).Error("error getting hive namespace")
+		return err
+	}
+	if v, ok := ns.Labels["openshift.io/cluster-monitoring"]; !ok || v != "true" {
+		// add the label
+		if ns.Labels == nil {
+			ns.Labels = map[string]string{}
+		}
+		ns.Labels["openshift.io/cluster-monitoring"] = "true"
+		if err := r.Client.Update(context.TODO(), ns); err != nil {
+			hLog.WithError(err).Error("error updating hive namespace label to enable monitoring")
+			return err
+		}
+	}
```

Comment:
Is this strictly necessary when running with monitoring for a namespace not prefixed with `openshift-`? Or is this just harmless in that case?

Reply:
definitely required for `openshift-`, will have to test if user-workload monitoring requires it. seems harmless to cover both cases by keeping it.